api/jobworker/v1/jobworker.proto

```diff
@@ -0,0 +1,52 @@
+syntax = "proto3";
+
+package jobworker.v1;
+
+service JobWorkerService {
+  rpc StartJob(StartJobRequest) returns (StartJobResponse) {}
+  rpc StopJob(StopJobRequest) returns (StopJobResponse) {}
+  rpc JobStatus(JobStatusRequest) returns (JobStatusResponse) {}
+  rpc StreamJobOutput(StreamJobOutputRequest) returns (stream StreamJobOutputResponse) {}
+}
+
+message StartJobRequest {
+  string command = 1;
+  repeated string args = 2;
+}
+
+message StartJobResponse {
+  string job_id = 1;
+}
+
+message StopJobRequest {
+  string job_id = 1;
+}
+
+message StopJobResponse {
+  int32 exit_code = 1;
```

Comment:
In my mind, yes, it had been synchronous, however, I'm not sure it should remain so.

From the library perspective, I could return a channel that's closed when the job completes. The client could then use the status command to get the exit code.

Then from the rpc perspective, I could return a stream that sends a message when the job completes. However, this may be a bit heavy handed. Streaming rpcs have their place, but for this, it may be better to have the client poll on status. I could go either way here and welcome any thoughts.

Reply:
updated. i like the async better anyway.